EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/CheckoutServiceImpl.java

```diff
@@ -62,17 +75,35 @@ public PurchaseResponse placeOrder(Purchase purchase) {
         Customer existingCustomer = customerRepository.findCustomerByFirstNameAndLastNameAndEmail(
                     customer.getFirstName(), customer.getLastName(), customer.getEmail());
 
+        boolean discountRateChanged = false;
+        String newDiscountRate = "";
+
         if(existingCustomer != null){
-            existingCustomer.add(order);
-            customerRepository.save(existingCustomer);
+            User userWithGivenCustomer = userRepository.findUserByCustomer(existingCustomer);
+            if(userWithGivenCustomer == null) {
+                existingCustomer.add(order);
+                customerRepository.save(existingCustomer);
+            } else {
+                DiscountRate discountRateBeforePurchase = DiscountRate.valueOf(userWithGivenCustomer.getDiscountRate());
+                existingCustomer.add(order);
+                customerRepository.save(existingCustomer);
+                DiscountRate discountRateAfterPurchase = loyaltyProgramService.getCustomerCurrentDiscountRate(existingCustomer);
+                if(!(discountRateAfterPurchase.equals(discountRateBeforePurchase))) {
```

Comment:
condition can be replaced with
`(!discountRateAfterPurchase.equals(discountRateBeforePurchase))`

Reply:
Done